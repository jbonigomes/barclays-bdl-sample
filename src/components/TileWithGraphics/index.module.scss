@import '../../index.scss';

.container {
  width: 300px;
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  font-family: 'ExpertSans-Light';

  img {
    margin-bottom: 16px;
  }

  button {
    width: 100%;
    margin-top: 24px;
  }

  .content {
    padding: 24px;
    display: flex;
    flex-direction: column;
  }

  @media (min-width: $breakpoint) {
    width: 444px;
    text-align: left;

    img {
      margin-bottom: 0;
    }

    button {
      width: auto;
      margin-top: 16px;
    }

    .content {
      flex-direction: row;
    }

    .left {
      display: flex;
      flex: 0 0 104px;
      align-items: center;
    }
  }
}
