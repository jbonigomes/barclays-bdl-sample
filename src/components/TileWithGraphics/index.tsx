import React from 'react'
import Tile from '../Tile'
import Lead from '../Lead'
import Button from '../Button'
import styles from './index.module.scss'

type TileWithGraphicsProps = {
  title: string,
  image: string,
  buttonText?: string,
  children: React.ReactNode,
  onClick?: (e: React.MouseEvent<HTMLElement>) => void,
}

const TileWithGraphics: React.SFC<TileWithGraphicsProps> = ({
  title,
  image,
  onClick,
  children,
  buttonText,
}: TileWithGraphicsProps) => (
  <div className={styles.container}>
    <Tile>
      <div className={styles.content}>
        <div className={styles.left}>
          <img src={image} alt="Tile graphic" />
        </div>
        <div>
          <Lead text={title} />
          <div>{children}</div>
          {onClick && buttonText && (
            <Button
              linkOnDesktop
              type="secondary"
              onClick={onClick}
              text={buttonText}
            />
          )}
        </div>
      </div>
    </Tile>
  </div>
)

export default TileWithGraphics
