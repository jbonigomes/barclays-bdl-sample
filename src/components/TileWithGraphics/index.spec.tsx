import React from 'react'
import renderer from 'react-test-renderer'

import TileWithGraphics from '.'

test('Match snapshot', () => {
  const props = {
    image: 'image',
    title: 'title',
  }

  const tree = renderer.create(
    <TileWithGraphics
      image={props.image}
      title={props.title}
    >
      Foo
    </TileWithGraphics>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('Match snapshot with button', () => {
  const props = {
    image: 'image',
    title: 'title',
    buttonText: 'buttonText',
    onClick: jest.fn(),
  }

  const tree = renderer.create(
    <TileWithGraphics
      image={props.image}
      title={props.title}
      buttonText={props.buttonText}
      onClick={props.onClick}
    >
      Foo
    </TileWithGraphics>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})
