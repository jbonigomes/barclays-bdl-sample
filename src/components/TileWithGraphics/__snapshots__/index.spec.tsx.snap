// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="tile"
  >
    <div
      className="content"
    >
      <div
        className="left"
      >
        <img
          alt="Tile graphic"
          src="image"
        />
      </div>
      <div>
        <div
          className="lead"
        >
          title
        </div>
        <div>
          Foo
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshot with button 1`] = `
<div
  className="container"
>
  <div
    className="tile"
  >
    <div
      className="content"
    >
      <div
        className="left"
      >
        <img
          alt="Tile graphic"
          src="image"
        />
      </div>
      <div>
        <div
          className="lead"
        >
          title
        </div>
        <div>
          Foo
        </div>
        <button
          className="secondary button  linkOnDesktop"
          onClick={[MockFunction]}
        >
          buttonText
        </button>
      </div>
    </div>
  </div>
</div>
`;
